sem/utils.py

```diff
@@ -708,9 +716,6 @@ def filter_logs(db,
     if sevirity_class is not None or components is not None:
         if isinstance(sevirity_class, str):
             sevirity_class = [sevirity_class]
-        for value in components.values():
-            if isinstance(value, str):
-                value = [value]
```

Comment:
separate the two ifs
```python
if sevirity_class is not None:
   # do stuff
if components is not None:
   # do stuff
```

Also, check if the input is of the correct type, e.g.,
```python
if isinstance(sevirity_class, str):
    sevirity_class = [sevirity_class]
elif isinstance(sevirity_class, str):
    # do nothing
else:
    raise TypeError() # add useful message
```

Reply:
> separate the two ifs

I clubbed them together because we have to apply the OR operation between the queries for both these filters, whereas we have to apply AND operator between the queries for all remaining filters.

Reply:
> Also, check if the input is of the correct type

Fixed in [404fd9a](https://github.com/signetlabdei/sem/pull/51/commits/404fd9acc107f8a5819a4dc2f8edb6d25e313fa0).